api/waste_item/adapters.py

```diff
@@ -56,6 +59,45 @@ def get_all_material_count(self) -> list:
         items = WasteItemModel.objects.values('material').annotate(count=models.Count('material'))
         return [StatsWasteItem(material=item['material'], count=item['count']) for item in items]
 
+    def list_by_user_id(self, user_id: str) -> list[WasteItem]:
+        items = WasteItemModel.objects.filter(user_id=user_id)
+        return [item.to_entity() for item in items]
+    
+    def getExcel(self, user_id: str):
+        items = self.list_by_user_id(user_id)
```

Comment:
This is wrong, you are using a port in another port, remember a port is meant to return entity items, but given you are on the adapter side, you dont have to limit yourself to entity items, you should instead use all the power of the Django ORM

```suggestion
    def get_excel(self, user_id: str):
        items = WasteItemModel.objects.filter(user_id=user_id)
```

you could even get more data all at once directly with the ORM, like:

```suggestion
    def get_excel(self, user_id: str):
        items = WasteItemModel.objects.filter(user_id=user_id).related("user")
```

Or whatever, in this case is not worth it, but is so you know, how is expected to work.

Also another thing I noticed is that this `get_excel` is defined here on the adapters, but I do not see his declaration on the core/ports interface